src/index.ts

```diff
@@ -453,19 +453,21 @@ export type Token = string | Key;
 function regexpToRegexp(path: RegExp, keys?: Key[]): RegExp {
   if (!keys) return path;
 
-  // Use a negative lookahead to match only capturing groups.
-  const groups = path.source.match(/\((?!\?)/g);
-
-  if (groups) {
-    for (let i = 0; i < groups.length; i++) {
-      keys.push({
-        name: i,
-        prefix: "",
-        suffix: "",
-        modifier: "",
-        pattern: ""
-      });
-    }
+  const groupsRegex = /\((?:\?<(.*?)>)?(?!\?)/g;
+
+  let index = 0;
+  let execResult = groupsRegex.exec(path.source);
+  while (execResult) {
+    keys.push({
+      // Use parenthesized substring match if available, index otherwise
+      name: execResult[1] || index,
+      prefix: "",
+      suffix: "",
+      modifier: "",
+      pattern: ""
+    });
+    index++;
```

Comment:
It would probably be good to stay consistent with the behavior in string tokens. In that method, we're only incrementing the counter each non-named group: https://github.com/pillarjs/path-to-regexp/blob/8b0ae94f9ada4705a6144d639088b058f52bf75d/src/index.ts#L189.

That would mean you probably want `execResult[1] || index++`. We should add a mixed test of named and unnamed groups to make sure it doesn't break.

Reply:
OK will do that. 

Does it mean that this `/\/(.+)\/(?<groupname>.+)\/(.+)/` called with `localhost/testRegex/testData/extraStuff` should result in `{"0":"testRegex","1":"extraStuff","groupname":"testData"}` ?